README.md

```diff
@@ -221,6 +221,14 @@ A governance platform where votes are not linked to Wallets. ZK can be used to f
 REST APIs can log IP addresses of the data source and the receiver. Building a decentralized layer for enabling privacy preserving APIs alongside rate-limiting with RLN can be quite impactful</details>
 
 
+<details>
+<summary><b>2FA with Waku</b></summary>
+<br>
+Build a decentralized 2FA using Waku and RLN membership sets. RLN can be used to determine membership of a user to a 2FA group and waku can be used to communicate 2FA codes generated.
```

Comment:
not sure if this method will preserve the privacy of the user, can you describe the user flow a little more?

Reply:
Haven't thought of detailed flow yet, but at a high level was thinking of following actors in the system. There maybe gaps in this and do suggest if you think of how they can be addressed.
1. Users who can register for a specific membership(which is kind of linked to 3rd party app). Haven't thought where memberships would be maintained. Maybe user can broadcast proof of registration in Waku network (which is stored by 2FA generator nodes).
2. 2FA generator nodes (some in the waku network), would generate 2FA based on user's request and communicate it to both user (2fa-front-end listening on Waku) and 3rd-party. Communication to user would be via Waku through relay broadcasting. Maybe 3rd-party comm also can be via Waku(but that sounds very complex)